flox/runtime/launcher/globus_compute.py

```diff
@@ -1,10 +1,25 @@
+from __future__ import annotations
+
+import typing
+
 import globus_compute_sdk
-from concurrent.futures import Future
-from typing import Any, Callable
 
-from flox.flock import FlockNode
 from flox.runtime.launcher.base import Launcher
 
+if typing.TYPE_CHECKING:
+    from concurrent.futures import Future
+    from typing import Any, Callable, TypeAlias, Union
+
+    from flox.flock import FlockNode
+
+    """
+    NodeCallable: TypeAlias = Union[
+        Callable[[FlockNode], Any],
+        Callable[[FlockNode, Any], Any],
+        Callable[[FlockNode, Any, ...], Any],
+    ]
+    """
```

Comment:
in https://github.com/nathaniel-hudson/FLoX/pull/27 I did this with a protocol, like so:

```
class LauncherFunction(Protocol):
    def __call__(self, node: FlockNode, *args: Any, **kwargs: Any) -> Any:
        ...
```

which can be used in the `submit` method of all launchers, and is compatible with `globus_compute.Executor.submit`

Reply:
Good suggestion. I'll make that change.